ramls/event.json

```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Event data model",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "id": {
+      "description": "UUID",
+      "type": "string"
+    },
+    "eventType": {
+      "description": "Unique Event type, human-readable String, not UUID",
+      "type": "string"
+    },
+    "metadata": {
+      "description": "Event metadata",
+      "type": "object",
+      "properties": {
+        "tenantId" : {
+          "description": "Tenant id",
+          "type": "string"
+        },
+        "eventTTL": {
+          "description": "Time-to-live (TTL) for event in minutes",
+          "type": "integer"
+        },
+        "correlationId": {
+          "description": "Id to track related events",
+          "type": "string"
+        },
+        "originalEventId": {
```

Comment:
How is this going to be used? How does it differ from the correlation ID?

Reply:
If it is a UUID, then please see comment about constraining UUID properties

Reply:
OriginalEventId is an id of the event from which a particular business flow started. CorrelationId is a property that would allow to trace the related events, like request and answer events.
And yes, I'll add the UUID constraints, thank you.

Reply:
Hmm, ok, thanks for the context. This is probably where my definition of events differs from other folks, as I think of them as notifications. The idea of request and response suggests to me more command or RPC semantics.

Reply:
Ok, thank you for the context.

I think my understanding of events is likely different to that modelled here. I think of them as notifications, and not a request / response pattern like RPC or commands

Reply:
No, your understanding is completely the same to ours. An event can be a simple notification, or it can cause a bunch of subsequent events, or you can expect to receive a response on that notification. There can be some period of time before you receive a feed back, may be it needs some you interaction. But any way using that correlationid you can collect all those events.

Reply:
@TarasSpashchenko Based upon what you described above, my understanding is somewhat different. Here is unlikely to be the best place for this convversation, so I shall step away.